src/hotspot/cpu/s390/macroAssembler_s390.cpp

```diff
@@ -6019,6 +6019,16 @@ void MacroAssembler::lightweight_lock(Register obj, Register temp1, Register tem
   // instruction emitted as it is part of C1's null check semantics.
   z_lg(mark, Address(obj, mark_offset));
 
+  if (UseObjectMonitorTable) {
+    // Clear cache in case fast locking succeeds.
```

Comment:
@xmas92: This comment sounds like it should only be done if fast locking succeeds. Why are we doing it regardless of that?

Reply:
Thanks for explaining! I'm ok with it and I have implemented it the same way for PPC64: https://github.com/openjdk/jdk/pull/20922